File: examples/hello_world.rb
# examples/hello_world.rb
# A simple example to test the tracer

class Greeter
  def initialize(name)
    @name = name
    @greet_count = 0
  end

  def greet
    @greet_count += 1
    message = "Hello, #{@name}!"
    puts message
    message
  end

  def greet_count
    @greet_count
  end
end

def main
  greeter = Greeter.new("World")
  greeter.greet
  greeter.greet
  puts "Greeted #{greeter.greet_count} times"
end

main

File: examples/hello_world.js
class Greeter {
  constructor(name) {
    this.name = name
    this.greetCount = 0
  }

  greet() {
    this.greetCount += 1
    const message = `Hello, ${this.name}!`
    console.log(message)
    return message
  }
}

async function main() {
  const greeter = new Greeter('World')
  greeter.greet()
  await Promise.resolve()
  await new Promise((resolve) => setTimeout(resolve, 1))
  greeter.greet()
  console.log(`Greeted ${greeter.greetCount} times`)
}

main()
